ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -80,16 +82,20 @@
   private final StoreKeyFactory factory;
   private final File bloomFile;
   private final StoreMetrics metrics;
-  // an approximation of the last modified time.
-  private final AtomicLong lastModifiedTimeSec;
   private final AtomicInteger numberOfItems;
+  private final Time time;
 
+  // an approximation of the last modified time.
+  private AtomicLong lastModifiedTimeSec;
   private MappedByteBuffer mmap = null;
   private IFilter bloomFilter;
   private int keySize;
   private int valueSize;
-  private Offset prevSafeEndPoint = null;
+  private short version;
+  // reset key refers to the first StoreKey that is added to the index segment
+  private Pair<StoreKey, PersistentIndex.IndexEntryType> resetKey = null;
```

Comment:
This was due to the intent of this attribute being added. Its sole purpose is to help in resetting the storeFindToken during replication if compaction happened and if the token is not found. Hence the naming